protocol/headers.go

```diff
@@ -42,114 +61,151 @@ type Headers struct {
 
 // CorrelationID returns the 'correlation-id' header value or empty string if not set.
 func (h *Headers) CorrelationID() string {
-	if h.Values[HeaderCorrelationID] == nil {
-		return ""
+	if value, ok := h.Values[HeaderCorrelationID]; ok && value != nil {
+		return value.(string)
 	}
-	return h.Values[HeaderCorrelationID].(string)
+	return ""
 }
 
-// Timeout returns the 'timeout' header value or empty string if not set.
-func (h *Headers) Timeout() string {
-	if h.Values[HeaderTimeout] == nil {
-		return ""
+// Timeout returns the 'timeout' header value or duration of 60 seconds if not set.
+func (h *Headers) Timeout() time.Duration {
+	if value, ok := h.Values[HeaderTimeout]; ok {
+		if duration, err := parseTimeout(value.(string)); err == nil {
+			return duration
+		}
+	}
+	return 60 * time.Second
```

Comment:
Is the client supposed to handle default values of the headers? Or by spec, the cloud backend is supposed to interpret missing values to their defaults? 
In general, the aim is always to send the smallest possible payload from the device - i.e. omit defaulting in this case.
Another topic is that if we add this default values handling here - it should be added everywhere for consistency. Otherwise - it's hard to use if it's randomly implemented.

Reply:
In this case the value is only returned/read and not set explicitly.

>  "In general, the aim is always to send the smallest possible payload from the device - i.e. omit defaulting in this case."

I'm totally agree with this statement, so it's good all getters/readers to return the default value no matter it's explicitly set or not (as it's done for the "response-required" value).
In order to make it consistent (and keep the payload compact), the setters/modifiers should remove the header if it is with its default value, i.e. no need to check a local variable before calling WithResponseRequired(required) - the Lib will do this for me.

For advanced users the Values map is exported so they can check any header existence or set it with its default value.